app/server/appsmith-interfaces/src/main/java/com/appsmith/external/pluginExceptions/AppsmithPluginError.java

```
@@ -9,6 +9,7 @@
 
     PLUGIN_ERROR(500, 5000, "PluginExecution failed with error {0}"),
     PLUGIN_STRUCTURE_ERROR(500, 5001, "Plugin failed to get structure with error {0}"),
+    PLUGIN_TIMEOUT_ERROR(500, 5002, "Plugin execution for query \"{0}\" timed out in {1}ms. Please increase timeout duration in your action settings or check your backend action endpoint."),
```

Could we throw a 504 error here?